crates/bevy_sprite/src/texture_slice/slicer.rs

```diff
@@ -198,12 +200,20 @@ impl TextureSlicer {
     ///
     /// * `rect` - The section of the texture to slice in 9 parts
     /// * `render_size` - The optional draw size of the texture. If not set the `rect` size will be used.
+    ///
+    /// # Panics
+    ///
+    /// Panics if any border values are bigger than `rect` half extents
     #[must_use]
-    pub(crate) fn compute_slices(
-        &self,
-        rect: Rect,
-        render_size: Option<Vec2>,
-    ) -> Vec<TextureSlice> {
+    pub fn compute_slices(&self, rect: Rect, render_size: Option<Vec2>) -> Vec<TextureSlice> {
+        let rect_size = rect.size() / 2.0;
+        if self.border.left >= rect_size.x
+            || self.border.right >= rect_size.x
+            || self.border.top >= rect_size.y
+            || self.border.bottom >= rect_size.y
+        {
+            panic!("TextureSlicer border has out of bounds values");
```

Comment:
Can we just return an empty vec or an Option instead of panicking?

Reply:
We can but then it will silently fail without any slicing applied. So at least a `log::error!` message ?

Reply:
Yeah, log an error and fail IMO.